### PR TITLE
packaging: Remove leftovers from the transition from TEE specific kernel / initrd / image to the "confidential" ones

### PR DESCRIPTION
**Pull request by fidencio**
Please, see each commit message for more details.

### CODE REVIEW

**Review by ChengyuZhu6 — APPROVED** [review]
LGTM, thanks @fidencio. One minor comment:  
You may need to update `kata_config_version`.

**Review by zvonkok — APPROVED** [review]
LGTM, thanks

**Comment by niteeshkd** [comment]
What about removing the following sev/snp entries under kernel in versions.yaml? 
```
   sev:
      description: "Linux kernel that supports SEV and SNP"
      url: "https://cdn.kernel.org/pub/linux/kernel/v5.x/"
      version: "v5.19.2"
    snp:
      description: "Linux kernel that supports AMD SEV-SNP for VMs"
      url: "https://cdn.kernel.org/pub/linux/kernel/v5.x/"
      version: "v5.19.2"
```

**Comment by fidencio** [comment]
> What about removing the following sev/snp entries under kernel in versions.yaml?

Very good call, and while doing this I also removed:
* kernel-tdx-experimental, as the confidential one is used instead
* image / initrd for tdx / sev, as the confidential one is used instead

**Review by GabyCT — APPROVED** [review]

File: tools/packaging/kernel/build-kernel.sh [hunk 1]
```diff
@@ -110,7 +110,7 @@ Options:
 	-t <hypervisor>	: Hypervisor_target.
 	-u <url>	: Kernel URL to be used to download the kernel tarball.
 	-v <version>	: Kernel version to use if kernel path not provided.
-	-x <type>	: Confidential guest protection type, such as sev, snp, tdx, or "confidential" (for all of those).
+	-x <type>	: All the confidential guest protection type for a specific architecture.
```
Comment (jodh-intel) [inline tools/packaging/kernel/build-kernel.sh:113]: Nit: This looks a bit odd as the only value currently is "confidential" so could we either remove the `<type>` arg here, or specify in the usage that the value must be "confidential"?

**Review by jodh-intel — APPROVED** [review]
Hi, @fidencio - just one nit, but overall...

lgtm

File: tools/packaging/kernel/build-kernel.sh (hunk 1 above)
Comment (fidencio) [reply tools/packaging/kernel/build-kernel.sh]: I've removed the type, thanks for the suggestion!

**Pull request merged**